logging/telemetryConfig.go

```diff
@@ -120,6 +130,22 @@ func (cfg TelemetryConfig) getInstanceName() string {
 	return fmt.Sprintf("%s:", pathHashStr[:16])
 }
 
+func SanitizeTelemetryString(input string, maxParts int) string {
+	parts := strings.SplitN(input, ":", maxParts)
+	for i := range parts {
+		// Limit to alphanumeric, and _-
+		if reg != nil {
+			parts[i] = reg.ReplaceAllString(parts[i], "")
+		}
+
+		// Limit length to support up to an ipv6 address
+		if len(parts[i]) > 39 {
+			parts[i] = parts[i][:39]
+		}
+	}
+	return strings.Join(parts, ":")
```

Comment:
add a limit on the length of the total combined output string.

Reply:
There's an implicit limit of `39 * parts + parts - 1` which I think is good since we don't have any strict sanitization requirements at the moment.